### PR TITLE
fix(shadow): Cloning a shape with shadow

### LINKED ISSUE #9711: [Bug]: Cloning a shape with shadow throws an error.
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta19

### In What environments are you experiencing the problem?

_No response_

### Node Version (if applicable)

None

### Link To Reproduction

https://codesandbox.io/p/devbox/fabric-clone-shadow-pps7l7?file=%2Fsrc%2Ftest-shadow-clone.tsx

### Steps To Reproduce

1. Create a rectangle A with shadow.
2. Call the clone method A.clone for B.
3. It throws an error.

### Expected Behavior

success

### Actual Behavior

image

[demo](https://codesandbox.io/p/devbox/fabric-clone-shadow-pps7l7?file=%2Fsrc%2Ftest-shadow-clone.tsx)

I found that it went to `[Shadow].fromObject().then` . Its return value is not a Promise.

### Error Message & Stack Trace

_No response_

### PR DESCRIPTION
**Pull request by zhe-he**
When cloning a shadow we expect the fromObject method to return a promise.
In this case it isn't and when a clone operation with a shadow is involved we have an error.

Adding async in front of the fromObject function fixes the problem

close #9711

### CODE REVIEW

File: src/Shadow.ts [hunk 1]
```diff
@@ -241,7 +241,7 @@ export class Shadow {
       : data;
   }
 
-  static fromObject(options: Partial<TClassProperties<Shadow>>) {
+  static async fromObject(options: Partial<TClassProperties<Shadow>>) {
```
Comment (asturur) [inline src/Shadow.ts:244]: oh good catch, this was a recent change.

File: src/Shadow.spec.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import '../jest.extend';
+import { Shadow } from './Shadow';
+
+describe('Shadow', () => {
+    it('fromObject', async () => {
+        const shadow = await Shadow.fromObject({});
+        expect(shadow).toMatchObjectSnapshot();
+        expect(shadow).toMatchObjectSnapshot({ includeDefaultValues: false });
+    });
+
```
Comment (asturur) [inline src/Shadow.spec.ts:10]: While this test works, it won't catch a regression.
I would suggest we add a test that tests that then is defined.

```ts
it('fromObject is a promise', async () => {
        const promise = Shadow.fromObject({});
        expect(promise.then).toBeDefined(); // <--- or whatever methods exist to test the existance of the then
    });
```

**Comment by asturur** [comment]
@zhe-he remember to run `npm run prettier:write` and add the changelog line.

**Comment by asturur** [comment]
( also i think you didn't add the snapshot object )

File: src/Shadow.spec.ts (hunk 2 above)
Comment (zhe-he) [reply src/Shadow.spec.ts:10]: done

**Comment by zhe-he** [comment]
> ( also i think you didn't add the snapshot object )（另外我认为您没有添加快照对象）

When I ran jest, I found an extra folder src/__snapshots__. I thought it didn't need to be added, so I filtered it out. Now I've added it back. 😢

**Comment by asturur** [comment]
Something broke in the normal unit tests, i ll check it up asap, sorry i can't now.

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
@zhe-he i m using your PR to fix the build_stats action have patience

**Comment by asturur** [comment]
yeeeee we got the stats running

**Comment by asturur** [comment]
The same principle can be applied to changelog and coverage reporting.

**Review by asturur — APPROVED** [review]

**Pull request merged**